core/tests/protractor_utils/users.js

```diff
@@ -102,9 +144,30 @@ var _completeSignup = async function(username) {
   await action.click('Agree to terms checkbox', agreeToTermsCheckbox);
 
   var registerUser = element(by.css('.protractor-test-register-user'));
+
+  // Signup page performs client side redirection which is known to cause
+  // "both angularJS testability and angular testability are undefined" flake.
+  // As suggested by protractor devs here (http://git.io/v4gXM), waiting for
+  // angular is disabled during client side redirects.
+  await browser.waitForAngularEnabled(false);
   await action.click('Register user button', registerUser);
 
+  // The await action.click only waits until button is clicked, it doesnot wait
```

Comment:
In general, please do not duplicate code. If you are doing similar things in multiple places, try to abstract the similar thing into a common utility function/service. This both enables/encourages reuse, and also helps prevent skew in the different individual implementations.

Again, ditto elsewhere.

Reply:
done